pydantic/networks.py

```diff
@@ -363,6 +363,13 @@ def import_email_validator() -> None:
         import email_validator
     except ImportError as e:
         raise ImportError('email-validator is not installed, run `pip install pydantic[email]`') from e
+    if getattr(email_validator, '__version__', '').partition('.')[0] == '2':
+        return
+    else:
+        input_email = 'testaddr@example.tld'
+        valid_email = email_validator.validate_email(input_email, check_deliverability=False)
+        if not hasattr(valid_email, 'normalized'):
+            raise ImportError('email-validator version >= 2.0 required')
```

Comment:
@commonism and @Viicos,

What are your thoughts on something like this:

```py
version_above_2_1 = getattr(email_validator, '__version__', '').partition('.')[0] == '2'

input_email = 'testaddr@example.tld'
valid_email = email_validator.validate_email(input_email, check_deliverability=False)
version_above_2_0 = hasattr(valid_email, 'normalized')

if not (version_above_2_0 or version_above_2_1):
    raise ImportError('email-validator version >= 2.0 required')
```

Certainly a tad bit less efficient, but I think the logic is much easier to understand. Happy to accept either way, but wanted to get your thoughts on this first.

Reply:
Why not using `importlib.metadata` instead? (we don't need the backport since Python 3.8):

```python
from importlib.metadata import version

if not version("email-validator").partition('.')[0] == '2':
    ...
```

(Unless I'm missing something?)

Reply:
Good call. Let's go with:

```suggestion
    if not version("email-validator").partition('.')[0] == '2':
        raise ImportError('email-validator version >= 2.0 required, run pip install -U email-validator')
```

Reply:
The check you provide is for email-validator >= 2.1 - won't make that much of a difference but allowing to uncrustify the version check to get rid of the v2.0 oddity.

v2.0 does not have __version__ (as v1.x), therefore __version__ alone is not enough and checking for normalized attribute would be required to test for v2.0.
If you intent to go with v2.1, change the requirements in the pyproject.toml to match the version check?

Reply:
@commonism,

The check works with `2.0.0`:

```py
import email_validator
from importlib.metadata import version

def import_email_validator() -> None:
    global email_validator
    try:
        import email_validator
    except ImportError as e:
        raise ImportError('email-validator is not installed, run `pip install pydantic[email]`') from e

    print(version("email-validator"))
    #> 2.0.0
    
    if not version("email-validator").partition('.')[0] == '2':
        raise ImportError('email-validator version >= 2.0 required, run pip install -U email-validator')

import_email_validator()
```

This doesn't raise an error. I don't get an error if I'm using 2.1 either. I do get an error if I'm using < v2

Reply:
You are right, importlib.metadata.version is pure magic.